components/root/root.tsx

```diff
@@ -358,6 +361,30 @@ export default class Root extends React.PureComponent<Props, State> {
         GlobalActions.redirectUserToDefaultTeam();
     }
 
+    trackUTMCampaign() {
+        const qs = new URLSearchParams(window.location.search);
+
+        // list of key that we want to track
+        const keys = ['utm_source', 'utm_medium', 'utm_campaign'];
+
+        const campaign = keys.reduce((acc, key) => {
+            if (qs.has(key)) {
+                const value = qs.get(key);
+                if (value) {
+                    acc[key] = value;
+                }
+                qs.delete(key);
+            }
+            return acc;
+        }, {} as Record<string, string>);
+
+        if (Object.keys(campaign).length > 0) {
+            trackEvent('utm_params', 'utm_params', campaign);
```

Comment:
Sidenote question, not an actual opinion that needs to be addressed:
Wondering if there's benefit for the category to be more broad like `attribution_capture`. Thinking if we ever want to have different event types for other attribution-related data that aren't UTM-specific (gclid, Google Client ID, gaexp, etc.) – and having all those event types living under that category. 0/5 since I'm ignorant when it comes to the db/data side and conventions already in place; like if it helps to be super specific with categories, or it doesn't really matter, or can be changed easily down the road.

Reply:
cc @sri-byte ^